### PR TITLE
Release notes for new release

### PR DESCRIPTION
**Pull request by skourta**
This pull request adds the release notes for Revision 315 of the OpenSearch Operator, documenting a major release with new features, upgrades, bug fixes, and compatibility updates. The release introduces support for Ubuntu 24.04, upgrades OpenSearch and Dashboards to version 2.19.4, and includes new authentication options and full Terraform support.

Key highlights from the release notes:

### Major features and upgrades
* Added support for Ubuntu 24.04 (Noble Numbat) for both OpenSearch and Dashboards.
* Upgraded OpenSearch and OpenSearch Dashboards to version 2.19.4.
* Introduced OAuth and JWT authentication for enhanced security and integration.
* Added full Terraform support for simple and large deployments, including charm and product modules.

### Bug fixes and stability improvements
* Fixed issues related to startup, failover bootstrapping, orchestrator demotion, and error handling in OpenSearch and Dashboards.
* Improved retry mechanisms and error isolation for cluster readiness and authentication flows.

### Other improvements and compatibility
* Migrated documentation to ReadTheDocs for better accessibility.
* Updated system requirements and compatibility matrix for hardware architecture, OpenSearch version

### CODE REVIEW

File: docs/reference/release-notes/revision-307.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Revision 307
+
+December 8, 2025
+
+This release of the OpenSearch Operator adds support for Ubuntu 24.04 (Noble Numbat), upgrades OpenSearch and OpenSearch Dashboards version to 2.19.4, and introduces significant features including OAuth and JWT authentication, and full Terraform support.
+
+[Charmhub](https://charmhub.io/opensearch) | [Deploy guide](https://canonical-charmed-opensearch.readthedocs-hosted.com/2/tutorial/) | [Upgrade instructions](https://canonical-charmed-opensearch.readthedocs-hosted.com/2/how-to/upgrade/) | [System requirements](https://canonical-charmed-opensearch.readthedocs-hosted.com/2/reference/system-requirements/)
```
Comment (izmalk) [inline docs/reference/release-notes/revision-307.md]: Suggestions:

1. Deploy guide should be a link to How to deploy guide, and not the tutorial
2. Upgrade instructions - let's link to the How to perform a minor upgrade guide

File: docs/reference/release-notes/revision-307.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+* Remove get-password [(PR \#130)](https://github.com/canonical/opensearch-dashboards-operator/pull/130)
+* [DPE-5909][DPE-4308] Support secure_authentication toggle: [(PR \#131)](https://github.com/canonical/opensearch-dashboards-operator/pull/131)
+
+
+## Compatibility 
+
+| Charm                         | Revision | Hardware architecture | OpenSearch version | Minimum Juju version |
+| :---------------------------- | :------- | :-------------------- | :----------------- | :------------------- |
+| Charmed OpenSearch            | 307      | AMD64                 | v2.19.4            | 3.5+                 |
+| Charmed OpenSearch Dashboards | 60       | AMD64                 | v2.19.4            | 3.5+                 |
```
Comment (izmalk) [inline docs/reference/release-notes/revision-307.md]: Suggestion: Let's add links to revision numbers here, somewhere to GitHub, most likely to the tagged commit. Also, we should add release artefacts here with respective links.

**Review by izmalk — CHANGES_REQUESTED** [review]

File: docs/reference/release-notes/revision-307.md (hunk 1 above)
Comment (skourta) [reply docs/reference/release-notes/revision-307.md]: Added

File: docs/reference/release-notes/revision-307.md (hunk 2 above)
Comment (skourta) [reply docs/reference/release-notes/revision-307.md]: Added

File: docs/reference/release-notes/revision-307.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+December 8, 2025
+
+This release of the OpenSearch Operator adds support for Ubuntu 24.04 (Noble Numbat), upgrades OpenSearch and OpenSearch Dashboards version to 2.19.4, and introduces significant features including OAuth and JWT authentication, and full Terraform support.
+
+[Charmhub](https://charmhub.io/opensearch) | [Deploy guide](https://canonical-charmed-opensearch.readthedocs-hosted.com/2/how-to/deploy/deploy-on-lxd/) | [Upgrade instructions](https://canonical-charmed-opensearch.readthedocs-hosted.com/2/how-to/upgrade/perform-a-minor-upgrade/) | [System requirements](https://canonical-charmed-opensearch.readthedocs-hosted.com/2/reference/system-requirements/) 
+
+## Charmed OpenSearch
+
+### Features
+* [DPE-6091] Add OAuth integration [(PR \#612)](https://github.com/canonical/opensearch-operator/pull/612)
```
Comment (izmalk) [inline docs/reference/release-notes/revision-307.md]: Question: Can we add links to the Jira tasks references in this file?

```suggestion
* [[DPE-6091](https://warthogs.atlassian.net/browse/DPE-6091)] Add OAuth integration [(PR \#612)](https://github.com/canonical/opensearch-operator/pull/612)
```

File: docs/reference/release-notes/revision-315.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+December 8, 2025
+
+This release of the OpenSearch Operator adds support for Ubuntu 24.04 (Noble Numbat), upgrades OpenSearch and OpenSearch Dashboards version to 2.19.4, and introduces significant features including OAuth and JWT authentication, and full Terraform support.
+
+[Charmhub](https://charmhub.io/opensearch) | [Deploy guide](https://canonical-charmed-opensearch.readthedocs-hosted.com/2/how-to/deploy/deploy-on-lxd/) | [Upgrade instructions](https://canonical-charmed-opensearch.readthedocs-hosted.com/2/how-to/upgrade/perform-a-minor-upgrade/) | [System requirements](https://canonical-charmed-opensearch.readthedocs-hosted.com/2/reference/system-requirements/) 
+
+## Charmed OpenSearch
+
+### Features
```
Comment (izmalk) [inline docs/reference/release-notes/revision-315.md:12]: chore: Make sure to have empty lines surrounding each heading (except for the first, H1, being on the first line). This is for Markdown parsing reliability.

```suggestion
### Features

```

File: docs/reference/release-notes/revision-307.md (hunk 3 above)
Comment (skourta) [reply docs/reference/release-notes/revision-307.md]: Added

File: docs/reference/release-notes/revision-315.md (hunk 4 above)
Comment (skourta) [reply docs/reference/release-notes/revision-315.md:12]: Added

File: docs/reference/release-notes/revision-307.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+* Rollback Cache use [(PR \#139)](https://github.com/canonical/opensearch-dashboards-operator/pull/139)
+
+### Other improvements 
+
+* [[DPE-6922]](https://warthogs.atlassian.net/browse/DPE-6922) Integrate TiCS [(PR \#176)](https://github.com/canonical/opensearch-dashboards-operator/pull/176)
+* [[DPE-4307]](https://warthogs.atlassian.net/browse/DPE-4307) HA process interrupt tests [(PR \#114)](https://github.com/canonical/opensearch-dashboards-operator/pull/114)
+* [[DPE-6876]](https://warthogs.atlassian.net/browse/DPE-6876) - Migration to spread [(PR \#178)](https://github.com/canonical/opensearch-dashboards-operator/pull/178)
+* [[DPE-6217]](https://warthogs.atlassian.net/browse/DPE-6217) Remove channel + revision mentions [(PR \#145)](https://github.com/canonical/opensearch-dashboards-operator/pull/145)
+* [[DPE-6536]](https://warthogs.atlassian.net/browse/DPE-6536) Improve terraform charm module [(PR \#164)](https://github.com/canonical/opensearch-dashboards-operator/pull/164)
+* [[DPE-6687]][[DPE-6690]][[DPE-6691]](https://warthogs.atlassian.net/browse/DPE-6687) Extend TF charm modules [(PR \#168)](https://github.com/canonical/opensearch-dashboards-operator/pull/168)
```
Comment (izmalk) [inline docs/reference/release-notes/revision-307.md]: Question: This and the next line have multiple square bracket elements. They are not rendering correctly due to absence of links. Do you want to have them as separate items or combine them in a single set of brackets with a separator, like a comma?

File: docs/reference/release-notes/revision-307.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+This release of the OpenSearch Operator adds support for Ubuntu 24.04 (Noble Numbat), upgrades OpenSearch and OpenSearch Dashboards version to 2.19.4, and introduces significant features including OAuth and JWT authentication, and full Terraform support.
+
+[Charmhub](https://charmhub.io/opensearch) | [Deploy guide](https://canonical-charmed-opensearch.readthedocs-hosted.com/2/how-to/deploy/deploy-on-lxd/) | [Upgrade instructions](https://canonical-charmed-opensearch.readthedocs-hosted.com/2/how-to/upgrade/perform-a-minor-upgrade/) | [System requirements](https://canonical-charmed-opensearch.readthedocs-hosted.com/2/reference/system-requirements/) 
+
+## Charmed OpenSearch
+
+### Features
+
+* [[DPE-6091]](https://warthogs.atlassian.net/browse/DPE-6091) Add OAuth integration [(PR \#612)](https://github.com/canonical/opensearch-operator/pull/612)
```
Comment (izmalk) [inline docs/reference/release-notes/revision-307.md]: Todo: All the links here look rather weird, because square brackets should not be part of the link. Please adjust the syntax.

For example:

```suggestion
* [[DPE-6091](https://warthogs.atlassian.net/browse/DPE-6091)] Add OAuth integration [(PR \#612)](https://github.com/canonical/opensearch-operator/pull/612)  
```

If you need more examples, see the [Kafka release notes](https://github.com/canonical/kafka-operator/edit/3/edge/docs/reference/release-notes/revision-205.md).

File: docs/reference/release-notes/revision-307.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+* Remove get-password [(PR \#130)](https://github.com/canonical/opensearch-dashboards-operator/pull/130)
+
+
+## Compatibility 
+
+| Charm                         | Revision                                                                     | Hardware architecture | OpenSearch version | Minimum Juju version | Artifacts                                                       |
+| :---------------------------- | :--------------------------------------------------------------------------- | :-------------------- | :----------------- | :------------------- | --------------------------------------------------------------- |
+| Charmed OpenSearch            | [307](https://github.com/canonical/opensearch-operator/tree/rev307)          | AMD64                 | v2.19.4            | 3.5+                 | Snap: [revision 98](https://snapcraft.io/opensearch)            |
+| Charmed OpenSearch Dashboards | [60](https://github.com/canonical/opensearch-dashboards-operator/tree/rev60) | AMD64                 | v2.19.4            | 3.5+                 | Snap: [revision 54](https://snapcraft.io/opensearch-dashboards) |
```
Comment (izmalk) [inline docs/reference/release-notes/revision-307.md]: Question: Do you think we can and/or should add a link to the OpenSearch version number to link to the upstream release notes? Like this one - https://github.com/opensearch-project/opensearch-build/blob/main/release-notes/opensearch-release-notes-2.19.4.md

```suggestion
| Charmed OpenSearch            | [307](https://github.com/canonical/opensearch-operator/tree/rev307)          | AMD64                 | [v2.19.4](https://github.com/opensearch-project/opensearch-build/blob/main/release-notes/opensearch-release-notes-2.19.4.md)            | 3.5+                 | Snap: [revision 98](https://snapcraft.io/opensearch)            |
| Charmed OpenSearch Dashboards | [60](https://github.com/canonical/opensearch-dashboards-operator/tree/rev60) | AMD64                 | [v2.19.4](https://github.com/opensearch-project/opensearch-build/blob/main/release-notes/opensearch-release-notes-2.19.4.md)            | 3.5+                 | Snap: [revision 54](https://snapcraft.io/opensearch-dashboards) |
```

**Review by izmalk — DISMISSED** [review]
LGTM, but could be better if we fix a couple of issues I have mentioned.

File: docs/reference/release-notes/revision-307.md (hunk 5 above)
Comment (skourta) [reply docs/reference/release-notes/revision-307.md]: I fixed the links. If you want to have them in one bracket we can do that as well.

File: docs/reference/release-notes/revision-307.md (hunk 6 above)
Comment (skourta) [reply docs/reference/release-notes/revision-307.md]: Fixed

File: docs/reference/release-notes/revision-307.md (hunk 7 above)
Comment (skourta) [reply docs/reference/release-notes/revision-307.md]: I added the links to the release notes pages on github. If you prefer them to go to the `.md` files let me know.

File: docs/reference/release-notes/revision-307.md (hunk 6 above)
Comment (izmalk) [reply docs/reference/release-notes/revision-307.md]: Can we do the same with the parentheses at the end, for the PR link? Sorry for not noticing earlier.

File: docs/reference/release-notes/revision-307.md (hunk 5 above)
Comment (izmalk) [reply docs/reference/release-notes/revision-307.md]: Up to you, I don't have any preference for this. Looks fine as it is now, IMHO.

File: docs/reference/release-notes/revision-307.md (hunk 6 above)
Comment (skourta) [reply docs/reference/release-notes/revision-307.md]: Done

File: docs/reference/release-notes/revision-315.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+* [[DPE-5460](https://warthogs.atlassian.net/browse/DPE-5460)] Restart data only nodes without lock ([PR \#460](https://github.com/canonical/opensearch-operator/pull/460))
+* [[DPE-5671](https://warthogs.atlassian.net/browse/DPE-5671)] Logic change in how check if CA rotation in complete in cluster ([PR \#486](https://github.com/canonical/opensearch-operator/pull/486))
+* only data node ignores lock if restarting ([PR \#762](https://github.com/canonical/opensearch-operator/pull/762))
+* Fixing Terraform Flaky Tests due to Limited storage on github runners ([PR \#765](https://github.com/canonical/opensearch-operator/pull/765))
+* Fix CA rotation edge case on small deployments ([PR \#767](https://github.com/canonical/opensearch-operator/pull/767))
+* Set blocked message on verify repository fail ([PR \#769](https://github.com/canonical/opensearch-operator/pull/769))
+* Clear missing relations ([PR \#763](https://github.com/canonical/opensearch-operator/pull/763))
+* remove verify repository from requirer error checks ([PR \#772](https://github.com/canonical/opensearch-operator/pull/772))
```
Comment (gatici) [inline docs/reference/release-notes/revision-315.md]: We can remove this as it is added in the snapshots-rework. We did not remove any old behaviour.

File: docs/reference/release-notes/revision-315.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+* [[DPE-7559](https://warthogs.atlassian.net/browse/DPE-7559)] Fix leader-elected hook for non-leader units ([PR \#661](https://github.com/canonical/opensearch-operator/pull/661))
+* [[DPE-6422](https://warthogs.atlassian.net/browse/DPE-6422)] Fix upgrade path with snap rev. 65 ([PR \#553](https://github.com/canonical/opensearch-operator/pull/553))
+* [[DPE-7076](https://warthogs.atlassian.net/browse/DPE-7076)] Data node fails to start ([PR \#671](https://github.com/canonical/opensearch-operator/pull/671))
+* [[DPE-7545](https://warthogs.atlassian.net/browse/DPE-7545)] Fix s3-uri-style handling ([PR \#654](https://github.com/canonical/opensearch-operator/pull/654))
+* [[DPE-5460](https://warthogs.atlassian.net/browse/DPE-5460)] Restart data only nodes without lock ([PR \#460](https://github.com/canonical/opensearch-operator/pull/460))
+* [[DPE-5671](https://warthogs.atlassian.net/browse/DPE-5671)] Logic change in how check if CA rotation in complete in cluster ([PR \#486](https://github.com/canonical/opensearch-operator/pull/486))
+* only data node ignores lock if restarting ([PR \#762](https://github.com/canonical/opensearch-operator/pull/762))
```
Comment (gatici) [inline docs/reference/release-notes/revision-315.md]:
```suggestion
* Only data node ignores lock if restarting ([PR \#762](https://github.com/canonical/opensearch-operator/pull/762))
```

File: docs/reference/release-notes/revision-315.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+* [[DPE-7560](https://warthogs.atlassian.net/browse/DPE-7560)] Remove wrong default base path for snapshots repositories ([PR \#658](https://github.com/canonical/opensearch-operator/pull/658))
+* [[DPE-6878](https://warthogs.atlassian.net/browse/DPE-6878)] Stabilize upgrade tests ([PR \#626](https://github.com/canonical/opensearch-operator/pull/626))
+* [[DPE-6787](https://warthogs.atlassian.net/browse/DPE-6787)] Add smoke test ([PR \#588](https://github.com/canonical/opensearch-operator/pull/588))
+* [[DPE-5667](https://warthogs.atlassian.net/browse/DPE-5667)] Move away from ops\_test's wait\_for\_idle to wait\_until ([PR \#479](https://github.com/canonical/opensearch-operator/pull/479))
+* Replace Discourse docs with a charm description on Charmhub ([PR \#760](https://github.com/canonical/opensearch-operator/pull/760))
+* Refactor code and verify repository after all units save credentials ([PR \#774](https://github.com/canonical/opensearch-operator/pull/774))
+* Rework Plugins ([PR \#775](https://github.com/canonical/opensearch-operator/pull/775))
+
```
Comment (gatici) [inline docs/reference/release-notes/revision-315.md:57]: We forgot to mention about snapshots rework: https://github.com/canonical/opensearch-operator/pull/718

File: docs/reference/release-notes/revision-315.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+## Charmed OpenSearch Dashboards
+
+### Features
+
+* [[DPE-5867](https://warthogs.atlassian.net/browse/DPE-5867)] Add terraform module  ([PR \#134](https://github.com/canonical/opensearch-dashboards-operator/pull/134))
+* [[DPE-6091](https://warthogs.atlassian.net/browse/DPE-6091)] Add oauth integration ([PR \#180](https://github.com/canonical/opensearch-dashboards-operator/pull/180))
+* [[DPE-6936](https://warthogs.atlassian.net/browse/DPE-6936)] - Update to 2.19.1 ([PR \#182](https://github.com/canonical/opensearch-dashboards-operator/pull/182))
+* [[DPE-7927](https://warthogs.atlassian.net/browse/DPE-7927)] - Disable TLS on relation broken ([PR \#213](https://github.com/canonical/opensearch-dashboards-operator/pull/213))
+* [[DPE-8018](https://warthogs.atlassian.net/browse/DPE-8018)] add JWT authentication ([PR \#217](https://github.com/canonical/opensearch-dashboards-operator/pull/217))
```
Comment (gatici) [inline docs/reference/release-notes/revision-315.md]:
```suggestion
* [[DPE-8018](https://warthogs.atlassian.net/browse/DPE-8018)] Add JWT authentication ([PR \#217](https://github.com/canonical/opensearch-dashboards-operator/pull/217))
```

File: docs/reference/release-notes/revision-315.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+* [[DPE-5832](https://warthogs.atlassian.net/browse/DPE-5832)][[DPE-6252](https://warthogs.atlassian.net/browse/DPE-6252)] Add 24.04 support ([PR \#138](https://github.com/canonical/opensearch-dashboards-operator/pull/138))
+* Add new Prometheus Rules using the python exporter ([PR \#159](https://github.com/canonical/opensearch-dashboards-operator/pull/159))
+* Bump version to 2.19.2 ([PR \#198](https://github.com/canonical/opensearch-dashboards-operator/pull/198))
+* Release 2.19.4 ([PR \#224](https://github.com/canonical/opensearch-dashboards-operator/pull/224))
+
+### Bug fixes 
+
+* [[DPE-7774](https://warthogs.atlassian.net/browse/DPE-7774)] Raise if error on install ([PR \#214](https://github.com/canonical/opensearch-dashboards-operator/pull/214))
+* [[DPE-7002](https://warthogs.atlassian.net/browse/DPE-7002)] Update opensearch health checks ([PR \#215](https://github.com/canonical/opensearch-dashboards-operator/pull/215))
+* [[DPE-7928](https://warthogs.atlassian.net/browse/DPE-7928)] block if we fail to get the provider info from the oauth relation ([PR \#218](https://github.com/canonical/opensearch-dashboards-operator/pull/218))
```
Comment (gatici) [inline docs/reference/release-notes/revision-315.md]:
```suggestion
* [[DPE-7928](https://warthogs.atlassian.net/browse/DPE-7928)] Block if we fail to get the provider info from the oauth relation ([PR \#218](https://github.com/canonical/opensearch-dashboards-operator/pull/218))
```

**Review by izmalk — DISMISSED** [review]

File: docs/reference/release-notes/revision-315.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+* [[DPE-5699](https://warthogs.atlassian.net/browse/DPE-5699)] - Update CI workflow versions, add juju 3.6, and remove build wrapper ([PR \#124](https://github.com/canonical/opensearch-dashboards-operator/pull/124))
+* Fix branch release on CI ([PR \#123](https://github.com/canonical/opensearch-dashboards-operator/pull/123))
+* Remove dependency breaking ccc-hub ([PR \#136](https://github.com/canonical/opensearch-dashboards-operator/pull/136))
+* Rollback Cache use ([PR \#139](https://github.com/canonical/opensearch-dashboards-operator/pull/139))
+
+### Other improvements 
+
+* [[DPE-6922](https://warthogs.atlassian.net/browse/DPE-6922)] Integrate TiCS ([PR \#176](https://github.com/canonical/opensearch-dashboards-operator/pull/176))
+* [[DPE-4307](https://warthogs.atlassian.net/browse/DPE-4307)] HA process interrupt tests ([PR \#114](https://github.com/canonical/opensearch-dashboards-operator/pull/114))
+* [[DPE-6876](https://warthogs.atlassian.net/browse/DPE-6876)] - Migration to spread ([PR \#178](https://github.com/canonical/opensearch-dashboards-operator/pull/178))
```
Comment (gatici) [inline docs/reference/release-notes/revision-315.md]:
```suggestion
* [[DPE-6876](https://warthogs.atlassian.net/browse/DPE-6876)] Migration to spread ([PR \#178](https://github.com/canonical/opensearch-dashboards-operator/pull/178))
```

File: docs/reference/release-notes/revision-315.md [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+* [[DPE-7002](https://warthogs.atlassian.net/browse/DPE-7002)] Update opensearch health checks ([PR \#215](https://github.com/canonical/opensearch-dashboards-operator/pull/215))
+* [[DPE-7928](https://warthogs.atlassian.net/browse/DPE-7928)] block if we fail to get the provider info from the oauth relation ([PR \#218](https://github.com/canonical/opensearch-dashboards-operator/pull/218))
+* [[DPE-6600](https://warthogs.atlassian.net/browse/DPE-6600)] Move away from `libjuju` and `add_machine` ([PR \#167](https://github.com/canonical/opensearch-dashboards-operator/pull/167))
+* [[DPE-6598](https://warthogs.atlassian.net/browse/DPE-6598)][[DPE-6786](https://warthogs.atlassian.net/browse/DPE-6786)] Update to 2.18 and fix CI ([PR \#169](https://github.com/canonical/opensearch-dashboards-operator/pull/169))
+* [[DPE-5699](https://warthogs.atlassian.net/browse/DPE-5699)] - Update CI workflow versions, add juju 3.6, and remove build wrapper ([PR \#124](https://github.com/canonical/opensearch-dashboards-operator/pull/124))
```
Comment (gatici) [inline docs/reference/release-notes/revision-315.md]:
```suggestion
* [[DPE-5699](https://warthogs.atlassian.net/browse/DPE-5699)] Update CI workflow versions, add juju 3.6, and remove build wrapper ([PR \#124](https://github.com/canonical/opensearch-dashboards-operator/pull/124))
```

**Review by gatici — COMMENTED** [review]
Thank you @skourta! Some minor things are commented.

**Review by gatici — DISMISSED** [review]

**Review by smethnani — APPROVED** [review]
Thank you @skourta!

**Review by gatici — APPROVED** [review]

**Review by izmalk — APPROVED** [review]
Looks good!
